lib/logstash/codecs/compress_spooler.rb

```diff
@@ -25,9 +31,10 @@ def encode(event)
     # MessagePack#pack which relies on pure Ruby object recognition
     @buffer << LogStash::Util.normalize(event.to_hash).merge(LogStash::Event::TIMESTAMP => event.timestamp.to_iso8601)
     # If necessary, we flush the buffer and get the data compressed
-    if @buffer.length >= @spool_size
+    if @buffer.length >= @spool_size || (@min_flush_time > 0 && (@last_flush + @min_flush_time < Time.now.to_i))
```

Comment:
what about wrapping this ```(@min_flush_time > 0 && (@last_flush + @min_flush_time < Time.now.to_i)``` condition in side a method? to we improve a bit readability, what do you think?